### DISCUSSION TITLE
Mapping openapi/v1.json fails on IIS when the api is in a subfolder

### DISCUSSION THREAD

**Discussion by KolRah** [scalar/scalar#4101] [Q&A]
Hi everyone

Sorry to send this question but I have an issue I cannot solve after many hours of research

When I'm on the development environment on my computer I can use localhost:5555/scalar/v1/ and it finds the openapi/v1.json file ok
If I install my api on IIS with an url like https://xxx.com/MyAPI/ I can correctly get  https://xxx.com/MyAPI/openapi/v1.json file but Scalar fails and skips the MyAPI part of the path: it looks for  https://xxx.com/scalar/v1

![image](https://github.com/user-attachments/assets/a855e9a1-f0ed-4086-8cbf-3b19e1b7c0fa)

Is there something I can do? I couldn't find a mean to solve this in a generic way (I have 40 apis, I don't want to hardcode the IIS subfolder name in the program.cs)

I use this configuration
![image](https://github.com/user-attachments/assets/d5c9271f-b4fe-470e-9783-873e43c168e4)

Thanks for your help
Raphael

**Comment by hanspagel** [comment]
Can you set the URL to openapi/v1.json (without a leading slash) and try again?

**Reply by KolRah** [reply]
Hi  Hans, thanks for your quick reply
Here's my code (with or without the swaggerUI thing) and it still does not work with the same error (maps to the root and not to the subsite)
![image](https://github.com/user-attachments/assets/c7982cc8-3da1-4b44-b337-e03286adc6aa)
Thanks

**Reply by KolRah** [reply]
Also I've installed the .net package for scalar.aspnet into my project but if I don't setup the WithCdnUrl, I have an error when the browser starts saying it looks for the scalar/v1/scalar.js file which does not exists. I have nothing as well in the .net project itself.
Am I missing something?

**Comment by xC0dex** [comment]
Hey @KolRah, do you have the information in the code (via IOptions or Configuration) about which subfolder the app is deployed in? If so, you could prefix the `OpenApiRoutePattern` property with the subfolder value.

**Reply by KolRah** [reply]
Hi @xC0dex , thanks I'll try on monday (today I'm off)
Basically I was looking for a way to have it work without parameterizing the path somewhere in the appsettings. Otherwise I'll do so but as stated, I have 40 apis and it'll be a pain to have to specify the folder every time I, for example, copy paste from Dev to Validation then to production  (didn't have this issue in the past with Swagger)
But of course if there is no other choice I'll do it
I keep you informed

**Reply by xC0dex** [reply]
I can fully understand this problem. We may have to solve this issue by using some JavaScript magic to resolve the sub path dynamically. However, this would be a breaking change, but definitely something worth considering for version 2.0.0.

**Reply by KolRah** [reply]
Thanks Justin
The OpenApiRoutePattern  does correct my issue
but yeah, avoiding declaring the application physical path in the appsettings to get the right /openapi/v1.json file may help a lot
I'm wondering why I'm the only one facing this issue
Thanks for your quick and kind help

**Reply by xC0dex** [reply]
You're not the only one! There were some other discussions/questions on GitHub and Discord. Other libraries like `Swashbuckle` don't have this issue because they use the same sub path prefix. I'll reach out to you once I created an issue/ TODO list for version 2.0 if that's ok.

**Reply by KolRah** [reply]
now my remaining issue is "my company proxy policy does not allow me to use a CDN" and I don't know where to put the scalar or standalone.js in my api project (and where to configure using it) so that it can be used.
(and when I have installed the Scalar.AspNetCore scalar works but I just have the dll and no other file for the js part)

**Reply by xC0dex** [reply]
Is there a need for you to override the CDN URL? By default, the `standalone.js` file will be provided under `/scalar/standalone.js`.

**Reply by KolRah** [reply]
if I remove the Cdn from the program.cs it screams for a scalar.js file in path /scalar/v1/scalar.js
![image](https://github.com/user-attachments/assets/cdffd8a1-bed8-4671-a75c-98998b645e4e)

**Reply by KolRah** [reply]
but if the url I call is /myApi/scalar/standalone.js then I have a correct display of the scalar page
but I cannot use it  

![image](https://github.com/user-attachments/assets/8a5ff060-74ca-46ab-99ff-bddaa4d5284c)

**Reply by xC0dex** [reply]
The trailing slash at the end of the URL is causing the issue. `v1` should be a resource and not a directory `v1/`. The slash introduces a new sub folder, so relative paths will be resolved to `/v1/` instead of `/scalar/`.

**Reply by KolRah** [reply]
ok thanks it word (display) but still cannot launch the methods as the dropdown for http/https 
![image](https://github.com/user-attachments/assets/1f1b9bed-9b19-43dc-ae88-cfdbf39c269e)

generates a blank url
![image](https://github.com/user-attachments/assets/bc49d12b-9c44-4b57-b2f0-076fad45daf4)

and the call link becomes https://%2A/myapi/endpoint/GetDataModel

**Reply by xC0dex** [reply]
Hm, this looks strange. The built-in OpenAPI document generator (shipped with the `Microsoft.AspNetCore.OpenApi` package) has some issues with generating the correct server URLs. I guess that's the problem you're experiencing. [Here](https://github.com/scalar/scalar/discussions/4025) is a discussion about this issue. There are some workarounds provided which may help you.
